compiler/rustc_span/src/symbol.rs

```diff
@@ -1512,9 +1512,12 @@ impl Ident {
         Ident::new(self.name, self.span.normalize_to_macro_rules())
     }
 
-    /// Convert the name to a `SymbolStr`. This is a slowish operation because
-    /// it requires locking the symbol interner.
-    pub fn as_str(self) -> SymbolStr {
+    /// Access the underlying string. This is a slowish operation because it
+    /// requires locking the symbol interner.
+    ///
+    /// Note that the lifetime of the return value is a lie. See
+    /// `Symbol::as_str()` for details.
+    pub fn as_str(&self) -> &str {
```

Comment:
The one thing is that SymbolStr was `!Send` and `!Sync`, while I think Symbol isn't. So now the return value can be (unsoundly) shared across threads.

Reply:
The api is already unsound before this PR. If you do `&*Box::leak(Box::new(sym.as_str()))` you will get an `&'static str`, which can outlive the interner. Sharing the return value between threads is fine for as long as the interner is alive. Both before and after this PR the `&str` should become inaccessible before the interner is dropped for as long as the symbol is not leaked. To be entirely sound the `.as_str()` method would need to be replaced with a `.with_str()` method taking a closure.

Reply:
Hmm, I forgot that it's possible to leak SymbolStr.